src/GafferUI/FPSGadget.cpp

```diff
@@ -0,0 +1,115 @@
+//////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (c) 2022, Image Engine Design Inc. All rights reserved.
+//
+//  Redistribution and use in source and binary forms, with or without
+//  modification, are permitted provided that the following conditions are
+//  met:
+//
+//      * Redistributions of source code must retain the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer.
+//
+//      * Redistributions in binary form must reproduce the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer in the documentation and/or other materials provided with
+//        the distribution.
+//
+//      * Neither the name of John Haddon nor the names of
+//        any other contributors to this software may be used to endorse or
+//        promote products derived from this software without specific prior
+//        written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "GafferUI/FPSGadget.h"
+
+#include "GafferUI/Style.h"
+#include "GafferUI/ViewportGadget.h"
+
+using namespace GafferUI;
+
+GAFFER_GRAPHCOMPONENT_DEFINE_TYPE( FPSGadget );
+
+FPSGadget::FPSGadget()
+	:	Gadget( defaultName<FPSGadget>() ), m_totalTime( 0 )
+{
+}
+
+FPSGadget::~FPSGadget()
+{
+}
+
+void FPSGadget::renderLayer( Gadget::Layer layer, const Style *style, Gadget::RenderReason reason ) const
+{
+	if( layer != Layer::Main )
+	{
+		return;
+	}
+
+	if( isSelectionRender( reason ) )
+	{
+		return;
+	}
+
+	static std::chrono::time_point<std::chrono::high_resolution_clock> lastTime;
+	static bool initialized = false;
```

Comment:
Perhaps `std::optional<time_point>`? That makes it impossible to use an uninitialised value, and is more explicit than a separate `initialized` variable (which could refer to anything).